### DISCUSSION TITLE
How to Convert VASP Output Files to MACE XYZ Format for Training and Testing Sets

### DISCUSSION THREAD

**Discussion by obaica** [ACEsuit/mace#457] [Q&A] [answered]
Hi MACE Developers,

I am using VASP software for my computational materials simulations, and I am interested in converting the output files from VASP into the MACE XYZ format for creating training and testing datasets. I would like to know if there is an existing functionality in MACE that facilitates this conversion, or if I need to write custom scripts to achieve this.

Any guidance or references to relevant documentation would be greatly appreciated.

Thank you for your support.

Best regards,

**Comment by ilyes319** [comment, marked as answer]
Hi @obaica,

You would want to look at the ASE documentation for the VASP interface: https://wiki.fysik.dtu.dk/ase/ase/calculators/vasp.html. If you do not want to use this interface, then you will have to code your own script.

**Reply by obaica** [reply]
Hi @ilyes319 ,

Thank you for your prompt response.

Yes, I am aware that pymatgen in conjunction with ASE can facilitate the conversion of VASP outputs to XYZ format. However, I would like to know if there are any specific requirements or important information that needs to be included in the XYZ format for creating training datasets for MACE.

Does the MACE documentation provide any guidelines or specifications regarding this?

Thank you for your assistance.

Best regards,

**Reply by ilyes319** [reply]
You can read the documentation here: https://mace-docs.readthedocs.io/en/latest/guide/training.html

MACE will read any xyz file as long as you have specified the correct keys for forces (in the atoms.arrays) and energies (in the atoms.info). We recommend using "REF_forces" and "REF_energy" as you key name. Then you will have to pass that to the input script as `--energy_key="REF_energy"` and `--forces_key="REF_forces"`.

**Comment by gabor1** [comment]
Independently of pymatgen, you can use ASE to read VASP output files. ASE can output Extended XYZ files which are then read by MACE. Make sure (and verify manually) that as you write those few lines of python using ASE, you get the "free energy" output from VASP as your "energy" that gets written to the XYZ file. Verify that your stresses get converted correctly in terms of units. MACE uses the default ASE units for stress.
